### PR TITLE
feat(otlp): wired through instrumentation scope metadata as tags config key

### LINKED ISSUE #2025: Wire up scope metadata config and preserve OTLP attribute values in metric tags
Sub-issue of #2005.

## What this controls

`instrumentation_scope_metadata_as_tags: bool` on `OtlpMetricsTranslatorConfig`
(`lib/saluki-components/src/sources/otlp/metrics/config.rs:47`) controls whether the OTel
instrumentation scope's name/version are added as tags on emitted metrics.

## Current state

- Agent config key: `otlp_config.metrics.instrumentation_scope_metadata_as_tags`.
- Currently hard-coded to `true` — never overridden, falls through to
  `OtlpMetricsTranslatorConfig::default()` at the construction site
  (`lib/saluki-components/src/sources/otlp/mod.rs:193-195`).
- Flagged `"ignored in initial audit 2026-04-23"` in `schema_overlay.yaml`.
- No corresponding parsed field exists yet on `OtlpConfig`'s `metrics: MetricsConfig`
  (`lib/saluki-components/src/common/otlp/config.rs:118,125,162-168`).

## Work

- [ ] Add an `instrumentation_scope_metadata_as_tags` field to `MetricsConfig`, default `true` for
  parity with today's behavior.
- [ ] Pass the parsed value into `OtlpMetricsTranslatorConfig` at `sources/otlp/mod.rs:193-195` via
  a `.with_instrumentation_scope_metadata_as_tags(...)` builder call (or equivalent).
- [ ] Document the field per `AGENTS.md` guidelines.
- [ ] Add a config test asserting the value flows through end-to-end.

## Requirements Tracking
- OM_CONF_303

---

## Additional scope: preserve OTLP attribute values in metric tags

> The issue author has used AI to research and write this issue. Please adversarially verify accuracy when working on the issue.

### Background

ADP does not convert all OTLP attribute values into metric tags in the same way as the Agent. Scope attributes with non-string values are dropped, and datapoint attributes with bytes, arrays, or maps are dropped. Empty datapoint attribute values are emitted with no value instead of using the Agent's `n/a` placeholder. The resource-versus-datapoint collision behavior also needs to be verified against the Agent before changing it.

### Agent behavior

The Agent converts instrumentation-scope metadata in `pkg/opentelemetry-mapping-go/otlp/metrics/internal/instrumentationscope/metadata.go:29-38`:

```go
for k, v := range il.Attributes().Range {
    tags = append(tags, utils.FormatKeyValueTag(k, v.AsString()))
}
```

`pcommon.Value.AsString()` is used for every OTLP value type, including strings, booleans, integers, doubles, bytes, arrays, and maps. Scope name and version are also passed through `FormatKeyValueTag`.

Datapoint attributes use the same conversion in `pkg/opentelemetry-mapping-go/otlp/metrics/dimensions.go:79-88`:

```go
labels.Range(func(key string, value pcommon.Value) bool {
    v := value.AsString()
    tags = append(tags, utils.FormatKeyValueTag(key, v))
    return true
})
```

`FormatKeyValueTag` in `pkg/opentelemetry-mapping-go/otlp/metrics/internal/utils/tags.go:16-23` changes an empty value into `n/a`, so `region: ""` becomes `region:n/a`.

The cited Agent metrics path does **not** implement datapoint-over-resource key precedence. `Dimensions.AddTags` stores datapoint and resource tags in a `[]string`, and `serializerexporter` preserves the concatenated list. Different values for the same key can both be present. The map-overwrite behavior cited by OM_TAGR_209 is in a trace attribute helper, not this metrics path. Do not introduce a new collision rule without first correcting that requirement.

### ADP behavior

The affected code is in the `saluki-components` crate:

- `lib/saluki-components/src/sources/otlp/metrics/internal/instrumentationscope/mod.rs:12-22` emits scope name and version, but its only scope-attribute match arm is `Value::StringValue`; booleans, integers, doubles, bytes, arrays, and maps are silently dropped.
- `lib/saluki-components/src/sources/otlp/metrics/dimensions.rs:48-77` converts datapoint attributes only when they are `StringValue`, `BoolValue`, `IntValue`, or `DoubleValue`. The `_ => continue` arm drops bytes, arrays, and key-value lists. It also constructs `format!("{}:{}", kv.key, v_str)` directly, so empty strings become `key:`.
- `lib/saluki-components/src/sources/otlp/metrics/internal/utils/mod.rs:4-8` already has `format_key_value_tag`, including the `n/a` substitution, but `Dimensions::with_attribute_map` does not use it.

Implement a shared OTLP `AnyValue` stringification path with semantics matching the Agent's `pcommon.Value.AsString()`, including the Agent's representation for bytes, arrays, and maps. Use it from both `tags_from_instrumentation_scope_metadata` and `Dimensions::with_attribute_map`, and pass all resulting values through `format_key_value_tag`.

`Dimensions.tags` is a `SharedTagSet`. `with_attribute_map` creates a datapoint `TagSet` and then calls `extend_from_shared(&self.tags)` to append resource and scope tags. `SharedTagSet` avoids repeated immutable chunks, while `TagSet` removes only exact duplicate tag strings; neither removes a resource tag when a datapoint tag has the same key. ADP therefore currently matches the Agent's metrics behavior for collisions. OM_TAGR_209 should be corrected or explicitly resolved rather than implemented as key-based precedence in this issue.

### Additional requirements tracking

- OM_TAGS_417
- OM_TAGS_784
- OM_TAGR_209

### Additional definition of done

- [ ] Add a red correctness test for all OTLP scope and datapoint attribute types, including bytes, arrays, maps, and empty strings.
- [ ] Share OTLP value stringification across scope and datapoint tags; apply `n/a` handling.
- [ ] Add or update a collision test matching Agent behavior; defer key precedence until `OM_TAGR_209` is corrected.
- [ ] PR passes correctness tests and removes the comment claiming the Agent drops composite attributes.

### PR DESCRIPTION
**Pull request by lucastemb**
## Summary

Previously, `otlp_config.metrics.instrumentation_scope_metadata_as_tags` was harcoded to `true`. This PR aims to expose a configuration option that allows for this value to be toggled on or off. 

## Change Type
- [ ] Bug fix
- [x] New feature
- [ ] Non-functional (chore, refactoring, docs)
- [ ] Performance

## How did you test this PR?

Unit tests
## References

- Closes: #2025

### CODE REVIEW

File: lib/agent-data-plane-config/src/domains/otlp.rs [hunk 1]
```diff
@@ -48,6 +52,11 @@ pub struct Metrics {
     /// semantic-convention mappings that are always applied.
     pub resource_attributes_as_tags: bool,
 
+    /// Whether instrumentation scope name, version, and attributes are added as metric tags.
+    ///
+    /// Defaults to `true`.
+    pub instrumentation_scope_metadata_as_tags: bool,
```
[the remark this answers is not in the document]
Comment (lucastemb) [reply lib/agent-data-plane-config/src/domains/otlp.rs:59]: Addressed in [1a39dd3](https://github.com/DataDog/saluki/pull/2366/commits/1a39dd3a45fad73d7d0d7fb827f0236fb5e77132)

**Review by tobz — APPROVED** [review]
